jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoostRegressor.scala

```diff
@@ -54,46 +54,91 @@ class XGBoostRegressor (
   def this(xgboostParams: Map[String, Any]) = this(
     Identifiable.randomUID("xgbr"), xgboostParams)
 
+  XGBoostToMLlibParams(xgboostParams)
+
   def setWeightCol(value: String): this.type = set(weightCol, value)
 
   def setBaseMarginCol(value: String): this.type = set(baseMarginCol, value)
 
-  private def fromXGBParamMapToParams(): Unit = {
-    for ((paramName, paramValue) <- xgboostParams) {
-      params.find(_.name == paramName) match {
-        case None =>
-        case Some(_: DoubleParam) =>
-          set(paramName, paramValue.toString.toDouble)
-        case Some(_: BooleanParam) =>
-          set(paramName, paramValue.toString.toBoolean)
-        case Some(_: IntParam) =>
-          set(paramName, paramValue.toString.toInt)
-        case Some(_: FloatParam) =>
-          set(paramName, paramValue.toString.toFloat)
-        case Some(_: Param[_]) =>
-          set(paramName, paramValue)
-      }
-    }
-  }
+  // setters for general params
+  def setNumRound(value: Int): this.type = set(numRound, value)
 
-  fromXGBParamMapToParams()
+  def setNumWorkers(value: Int): this.type = set(numWorkers, value)
 
-  private[spark] def fromParamsToXGBParamMap: Map[String, Any] = {
-    val xgbParamMap = new mutable.HashMap[String, Any]()
-    for (param <- params) {
-      if (isDefined(param)) {
-        xgbParamMap += param.name -> $(param)
-      }
-    }
-    val r = xgbParamMap.toMap
-    r
-  }
+  def setNthread(value: Int): this.type = set(nthread, value)
```

Comment:
can we have deduplicate these duplicate functions like setThread() in XGBRegressor and XGBClassifier?

Reply:
Putting setter function here is for Java compatibility.